src/pymor/algorithms/vkoga/vkoga.py

```diff
@@ -0,0 +1,358 @@
+        else:
+            self._centers = np.vstack([self._centers, mu])
+            self._centers_idx = np.concatenate([self._centers_idx, np.array([idx_in_X], dtype=int)])
+
+    def _update_newton_basis(self, idx_in_X):
+        """Append Xi as the last block to the Newton basis V."""
+        if self._V is None:
+            self._V = self._K_X_mu / np.sqrt(self._power2[idx_in_X])
+        else:
+            Xi = self._K_X_mu - (self._V[idx_in_X] @ self._V.T).reshape(-1, 1)
+            Xi = Xi / np.sqrt(self._power2[idx_in_X])
+            self._V = np.concatenate([self._V, Xi], axis=1)
+
+    def _update_cholesky_inverse(self, idx_in_X):
+        """Extend the inverse of the Cholesky matrix by the new block."""
+        c_nn = 1 / self._V[idx_in_X, -1]
+        if self._C is None:
+            self._C = np.atleast_2d(c_nn)
+        else:
+            self._C = np.block([[self._C, np.zeros((self._C.shape[0], 1))],
+                                [- c_nn * self._V[idx_in_X, :-1] @ self._C, c_nn]])
+
+    def _update_power_function_evals(self):
+        """Incrementally update self._power2 after adding new center."""
+        # update power2: p_{n+1}^2(x) = p_n^2(x) - norms
+        norms = self._V[:, -1] ** 2
+        self._power2 = np.maximum(self._power2 - norms, 0)
+
+
+class VKOGAEstimator(BasicObject):
+    """Scikit-learn-style estimator using the :class:`VKOGASurrogate`.
+
+    The estimator uses the :func:`~pymor.algorithms.greedy.weak_greedy` in its `fit`-method
+    to select centers according to the given criterion.
+
+    The algorithm is described in :cite:`WH13` and :cite:`SH21`.
+
+    Parameters
+    ----------
+    kernel
+        Kernel to use in the estimator.
+    criterion
+        Selection criterion for the greedy algorithm. Possible values are `'fp'`, `'f'` and `'p'`.
+    max_centers
+        Maximum number of selected centers in the greedy algorithm.
+    tol
+        Tolerance for the weak greedy algorithm.
+    reg
+        Regularization parameter for the kernel interpolation.
+    """
+
+    @defaults('criterion', 'max_centers', 'tol', 'reg')
+    def __init__(self, kernel, criterion='fp', max_centers=20, tol=1e-6, reg=1e-12):
+        self.__auto_init(locals())
+        self._surrogate = None
+
+    def fit(self, X, Y):
+        """Fit VKOGA surrogate using pyMOR's weak greedy algorithm.
+
+        Parameters
+        ----------
+        X
+            Training inputs.
+        Y
+            Training targets.
+
+        Returns
+        -------
+        The trained estimator.
+        """
+        X = np.asarray(X)
+        Y = np.asarray(Y)
+
+        # instantiate surrogate
+        surrogate = VKOGASurrogate(kernel=self.kernel, X_train=X, F_train=Y, criterion=self.criterion, reg=self.reg)
+
+        # use X as training set in the weak greedy algorithm
+        result = weak_greedy(surrogate, X, atol=self.tol, max_extensions=self.max_centers)
```

Comment:
If I understand correctly what is going on here, then you are passing a numpy array for `weak_greedy's` `training_set` argument. This makes me a bit uneasy, because I always thought of this as a set/list of `Mus`. Now, the docstring actually does not say anything about what the training set should be. So we might make this a `Sequence[Any]` or something like that. If we see no benefit in assuming any structure of the training data in `weak_greedy`, I'm in favor of it, and we should add some like to the documentation of `weak_greedy`.

Reply:
You are absolutely right. In my humble opinion, this is currently the nice thing about the VKOGA implementation: We just use pyMOR's `weak_greedy` and do not mimic a greedy algorithm within the `fit`-method. So if you are fine with that, it would (from my perspective) be great to stick to this structure and maybe extend the documentation of the `weak_greedy`.

Reply:
Great, I just added another sentence to the docs of `weak_greedy`. I hope this is fine.